client/data/deposits/selectors.js

```diff
@@ -36,6 +36,76 @@ export const getDepositsOverviewError = ( state ) => {
 	return DepositsOverview.error;
 };
 
+/**
+ * Prepares and returns all deposits' overviews from the state.
+ *
+ * The API returns this in the form of separate arrays
+ * for each balance/deposit type, each of them containing
+ * a separate entry for each currency.
+ *
+ * This selector will return the account, as well as an
+ * array with all of the neccessary details grouped by currency.
+ *
+ * Check the `AccountOverview.OverviewsResponse` declaration
+ * for the shape of the returned object.
+ *
+ * @param {Object} state Current wp.data state.
+ * @return {Object} A complex object, containing all neccessary overviews.
+ */
+export const getAllDepositsOverviews = ( state ) => {
+	const DepositsOverview = getDepositsState( state ).overviews || {};
+
+	// Return an empty skeleton if data has not been loaded yet.
+	if ( ! DepositsOverview.data ) {
+		return {
+			account: null,
+			currencies: [],
+		};
+	}
+
+	const { deposit, balance, account } = DepositsOverview.data;
+	const groups = {
+		lastPaid: deposit.last_paid,
+		nextScheduled: deposit.next_scheduled,
+		pending: balance.pending,
+		available: balance.available,
+	};
+
+	const currencies = {};
+	for ( const [ key, values ] of Object.entries( groups ) ) {
+		values.forEach( ( value ) => {
+			const { currency } = value;
+
+			if ( ! currencies[ currency ] ) {
+				currencies[ currency ] = {
+					currency,
+					lastPaid: null,
+					nextScheduled: null,
+					pending: null,
+					available: null,
+				};
+			}
+
+			// There will be a single deposit/balance per currency, no arrays here.
+			currencies[ currency ][ key ] = value;
+		} );
+	}
```

Comment:
Just a thought: can we move this logic to resolver so that this computation is not done each time the selector is called, or maybe we can memorize the result of this selector. What do you think?

Reply:
That's good advice. We could potentially also use `useEffect` to make it a bit more React-style.

I don't think it's worth it though:
- I'd like to keep this logic in the selector, and that leaves memorization as the only choice.
- Looking at the design of the page, this selector will be called few times.
- There is not a lot of computation in the selector, and the dataset is and will be pretty small in most cases.

Do you agree?

Reply:
I agree, make sense, though can you please add a comment on the top indicating that we can add memorization later on if required.

Reply:
Alright. Done here: https://github.com/Automattic/woocommerce-payments/pull/1892/commits/ce426cc54a41bdd915a32a2e72d89eea4cc72b97#diff-35978bc52ccca1b99c8528f6fe263d7c46de7754ab5a0ae54f48fcecfa0ee11fR76